### PR TITLE
pallet-revive: Fix docs.rs

### PR DESCRIPTION
**Pull request by athei**
- Fixed failing docs.rs build for `pallet-revive-uapi` by fixing a wring attribute in the manifest (we were using `default-target` instead of `targets`)
-  Removed the macros defining host functions because the cfg attributes introduced in #6866 won't work on them
- Added an docs.rs specific attribute so that the `unstable-hostfn` feature tag will show up on the functions that are guarded behind it.

### CODE REVIEW

**Comment by athei** [comment]
bot fmt

**Review by jarkkojs — CHANGES_REQUESTED** [review]
I don't see prdoc entry, should this have one? Description could explain migration to unstable_hostfn, which would then also help to review the PR overall.

I mean structure of this is like:

1. There's actually only one quite scoped change.
2. Rest "ripples" from that.

So explaining 1 would help others to understand this.

**Comment by athei** [comment]
Ooops. Yeah we need a prdoc in order to bump the pallets on release.

File: substrate/frame/contracts/uapi/Cargo.toml [hunk 1]
```diff
@@ -21,7 +21,7 @@ codec = { features = [
 ], optional = true, workspace = true }
 
 [package.metadata.docs.rs]
-default-target = ["wasm32-unknown-unknown"]
+targets = ["wasm32-unknown-unknown"]
```
[the remark this answers is not in the document]
Comment (athei) [reply substrate/frame/contracts/uapi/Cargo.toml:24]: Have you read the PR description? It's the first point there.

**Comment by jarkkojs** [comment]
> Ooops. Yeah we need a prdoc in order to bump the pallets on release. Generated from PR description.

Hmm... there's a way to generate it? :-) I've created the file manually.

File: prdoc/pr_6896.prdoc [hunk 2]
```diff
@@ -0,0 +1,16 @@
+title: 'pallet-revive: Fix docs.rs'
+doc:
+- audience: Runtime Dev
+  description: |-
+    - Fixed failing docs.rs build for `pallet-revive-uapi` by fixing a wring attribute in the manifest (we were using `default-target` instead of `targets`)
```
Comment (jarkkojs) [inline prdoc/pr_6896.prdoc:5]: wring i.e. wiring?

Comment (jarkkojs) [reply prdoc/pr_6896.prdoc:5]: this is my only complaint, otherwise makes sense

**Review by jarkkojs — APPROVED** [review]

**Pull request merged**